x-pack/plugins/enterprise_search/server/routes/__mocks__/router.mock.ts

```diff
@@ -14,13 +19,24 @@ import { IRouter, RequestHandlerContext, RouteValidatorConfig } from 'src/core/s
 type methodType = 'get' | 'post' | 'put' | 'patch' | 'delete';
 type payloadType = 'params' | 'query' | 'body';
 
+interface IMockRouterProps {
+  method: methodType;
+  payload: payloadType;
+}
+interface IMockRouterRequest {
+  body?: object;
+  query?: object;
+  params?: object;
+}
+type TMockRouterRequest = KibanaRequest | IMockRouterRequest;
+
 export class MockRouter {
-  public router: jest.Mocked<IRouter>;
+  public router!: jest.Mocked<IRouter>;
```

Comment:
Just curious, as I've never seen this before, but what is the bang`!` for?

Reply:
Typescript was yelling about this potentially being undefined 😬 The error in question was:

> Property 'router' has no initializer and is not definitely assigned in the constructor.

Which is incorrect, because I **am** assigning `this.router` in `constructor()`... it's just being assigned within `this.createRouter()`, but for whatever reason Typescript isn't parsing that correctly.

To be honest lol I mostly fixed it by clicking the yellow suggest bulb in VSCode:

<img width="608" alt="" src="https://user-images.githubusercontent.com/549407/83697142-177cb800-a5b3-11ea-96aa-d468cc2e279d.png">

But afterwards did some digging and it's called the [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) which is basically forcing Typescript to not complain about it possibly being undefined/null :trollface: